### ISSUE TITLE
Faraday 1.x.x compability

### ISSUE THREAD

**Issue by ayanko** [amro/gibbon#302]
https://github.com/lostisland/faraday/blob/master/UPGRADING.md

So we need on

https://github.com/amro/gibbon/blob/master/lib/gibbon/api_request.rb#L112

```diff
-   if error.is_a?(Faraday::ClientError) && error.response
+   if (error.is_a?(Faraday::ClientError) || error.is_a?(Faraday::ServerError) ) && error.response
```

**Comment by amro** [comment]
Thanks very much for pointing this out. I can take a look and test this on Friday

**Comment by amro** [comment]
Thanks again for the callout and contribution @ayanko. I've just released Gibbon 3.4.0 w/ your change in place to address this. I did need to make some minor config changes to accommodate testing Ruby 3.0.0 on CI.
